### PR TITLE
Support Tabs vs Spaces in notebook editors

### LINKED ISSUE #108675: Cannot pick tabs vs spaces in notebook
No a "choose intentation type" entry in status bar when editing in a notebook. I have a notebook with tab indentation but it seems theres some global spaces setting that overrides it... even pasting a literal tab charachter inserts spaces :/ 

recording (10)

### PR DESCRIPTION
**Pull request by Yoyokrazy**
Fixes: #108675

Currently you can control the indentation modes for notebook editors via the setting:
`"notebook.editorOptionsCustomizations": {}`

For example:
```json
"notebook.editorOptionsCustomizations": {
	"editor.detectIndentation": false,
	"editor.indentSize": "tabSize",
	"editor.tabSize": 16,
	"editor.insertSpaces": true,
},
```

### CODE REVIEW

**Comment by Yoyokrazy** [comment]
Todo: 
- [ ] hook up command for statusbar entry to change indentation types

File: src/vs/workbench/contrib/notebook/browser/view/cellParts/codeCell.ts [hunk 1]
```diff
@@ -135,9 +136,25 @@ export class CodeCell extends Disposable {
...
+		templateData.editor.updateOptions(this._cellEditorOptions.getUpdatedValue(this.viewCell.internalMetadata, this.viewCell.uri));
+	}
+
+	private updateCodeCellOptions(templateData: CodeCellRenderTemplate) {
+		templateData.editor.updateOptions(this._cellEditorOptions.getUpdatedValue(this.viewCell.internalMetadata, this.viewCell.uri));
+
+		const cts = new CancellationTokenSource();
+		this._register({ dispose() { cts.dispose(true); } });
+		raceCancellation(this.viewCell.resolveTextModel(), cts.token).then(model => {
+			if (model) {
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/view/cellParts/codeCell.ts:150]: the model might be disposed here, so we might want to check `this.isDisposed`

File: src/vs/workbench/contrib/notebook/browser/view/cellParts/markupCell.ts [hunk 2]
```diff
@@ -174,9 +174,27 @@ export class MarkupCell extends Disposable {
...
+	private updateMarkupCellOptions(): any {
+		this.updateEditorOptions(this.cellEditorOptions.getUpdatedValue(this.viewCell.internalMetadata, this.viewCell.uri));
+
+		if (this.editor) {
+			this.editor.updateOptions(this.cellEditorOptions.getUpdatedValue(this.viewCell.internalMetadata, this.viewCell.uri));
+
+			const cts = new CancellationTokenSource();
+			this._register({ dispose() { cts.dispose(true); } });
+			raceCancellation(this.viewCell.resolveTextModel(), cts.token).then(model => {
+				if (model) {
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/view/cellParts/markupCell.ts:189]: Same here.

**Review by rebornix — APPROVED** [review]
Looks good to me. Let's make sure we have issues tracking https://github.com/microsoft/vscode/pull/204783/files#diff-4677a47fd6ba528332f9c7620a387a37bc7092af04591fdcb1a16701f8fe5951R97 and in this PR we already have two places parsing the indent* settings ourselves.

**Pull request merged**